### ISSUE TITLE
Recenter gyro

### ISSUE THREAD

**Issue by JapanYoshi** [EX-XDRiVER/Issues#6]
The gyro on my controller works, but there is no way to reregister the current orientation of the controller as the origin.

Wii Sports Resort's Chambara mode, as well as the 3D All-Stars port of Super Mario Galaxy, lets you recalibrate the gyro controls using a button. EX-XDRiVER needs this feature for gyro to be functional without having to unplug and replug the controller all the time.

**Comment by RocketLanterns** [comment]
I'm not sure if it's consistent across all controllers, but on steam deck, I've found that clicking the left stick recenters the gyro.

**Comment by tari-cat** [comment]
Hey! Sorry about the slow response.
Re-centering the gyroscope might not be fully possible given what we have to work with since we use Steam's gyroscope data directly. The 'left-stick press' bind just sets more of an offset to neutral.
Unplugging and plugging the controller back in is the only way to fully reset the gyroscope at the moment until we make our own gyroscope implementations.

**Comment by AlexisBlade2001** [comment]
Is there any mention of the left stick click inside the game? there should have a mention of it, I tried to recalibrate it through Steam's Input, but kept being set up incorrectly (even though I let the controller on a table), I just read this, pressed the button and it works properly now... maybe this action should also be an option inside the gyroscope section, while mentioning that the player can recalibrate in-game (if the gyro gets funky)

**Comment by JapanYoshi** [comment]
Yeah, I think “Press L3 to recenter the gyroscope” should be shown in the gyro settings menu, given that’s where people would naturally go to when they feel the gyro origin is wrong.

**Comment by tari-cat** [comment]
It is not currently mentioned. It may be mentioned in the future gyroscope rework once I end up figuring it out. At the moment the gyroscope isn't very cooperative when it comes to changes. It's purely what Steam sends at the current moment.
